### PR TITLE
[QC-547] Allow to store and retrieve any object with dictionary

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

**Comment by Barthelemy** [comment]
Depends on https://github.com/AliceO2Group/AliceO2/pull/5748

**Comment by Barthelemy** [comment]
I could not use a nice templated method because we have an interface and thus a vtable. Let me know if it is ok like that.

**Review by knopers8 — APPROVED** [review]
All seems fine, thank you. Please explain the argument, as in the comment.

File: Framework/include/QualityControl/DatabaseInterface.h [hunk 1]
```diff
@@ -59,6 +59,50 @@ class DatabaseInterface
...
+   * ```
+   *
+   * Note that we cannot have a more elegant templated signature due to the fact that it is a virtual method.
+   *
+   * @param typeInfo The type of the object.
+   * @param path The path where the object is to be found.
+   * @param metadata Key-values representing the metadata to filter out objects.
+   * @param timestamp Timestamp of the object to retrieve. If omitted, current timestamp is used.
+   * @param optional headers Map to be populated with the headers we received, if it is not null.
+   * @param optional etag from previous call
```
[the remark this answers is not in the document]
Comment (Barthelemy) [reply Framework/include/QualityControl/DatabaseInterface.h:96]: bad copy paste

**Comment by Barthelemy** [comment]
@TimoWilken @ktf The error message 
```
/mnt/mesos/sandbox/sandbox/qualitycontrol-dataflow/sw/SOURCES/QualityControl/build_QualityControl_o2-dataflow/0/Framework/src/MySqlDatabase.cxx:95:42: error: unused parameter 'obj' [-Werror=unused-parameter]
 void MySqlDatabase::storeAny(const void* obj, std::type_info const& typeInfo, std::string const& path, std::map<std::string, std::string> const& metadata,
```
Does not correspond to the code in the PR:
```
 void MySqlDatabase::storeAny(const void*, std::type_info const&, std::string const&, std::map<std::string, std::string> const&, std::string const&, std::string const&, long, long)
```

How is that possible ?

**Comment by Barthelemy** [comment]
@knopers8 do you see what is wrong with `o2-dataflow` ? I don't.

**Comment by knopers8** [comment]
Me neither.

**Comment by Barthelemy** [comment]
Force merging then.

**Pull request merged**